pkg/subctl/cmd/root.go

```diff
@@ -234,3 +240,31 @@ var nodeLabelBackoff wait.Backoff = wait.Backoff{
 	Factor:   1.2,
 	Jitter:   1,
 }
+
+func checkVersionMismatch(cmd *cobra.Command, args []string) error {
+	config, err := getRestConfig(kubeConfig, kubeContext)
+	exitOnError("The provided kubeconfig is invalid", err)
+
+	clientSet, err := submarinerclientset.NewForConfig(config)
+	exitOnError("Unable to get the Submariner config", err)
+
+	submariner, err := clientSet.SubmarinerV1alpha1().Submariners(OperatorNamespace).Get(submarinercr.SubmarinerName, metav1.GetOptions{})
+
+	if errors.IsNotFound(err) {
+		return nil
+	}
+	exitOnError("Error retrieving the Submariner resource", err)
```

Comment:
There's a `getSubmarinerResource` function you can use.

```suggestion
	submariner := getSubmarinerResource(config)

	if submariner == nil {
	    return nil
	}
```

Reply:
I tried to use it but the function exits in case of an error

Reply:
yes but you're also using `exitOnError` on L244 and L247.

Reply:
Oh, I've done my tests before this code was added in `getSubmarinerResource`
```
if errors.IsNotFound(err) {
		return nil
}
```
It should work fine now thanks to @sridhargaddam +1